espnet/nets/pytorch_backend/transducer/tdnn.py

```diff
@@ -49,40 +49,81 @@ def __init__(
             idim, odim, ctx_size, stride=stride, dilation=dilation
         )
 
+        if self.relu:
+            self.rel = torch.nn.ReLU()
+
         if self.batch_norm:
             self.bn = torch.nn.BatchNorm1d(odim)
 
         self.dropout = torch.nn.Dropout(p=dropout_rate)
 
-    def forward(self, xs, masks):
+    def forward(
+        self,
+        x_input: Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]],
+        masks: torch.Tensor,
+    ) -> Tuple[Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]], torch.Tensor]:
         """Forward TDNN.
 
         Args:
-            xs (torch.Tensor): input tensor (B, seq_len, idim)
-            masks (torch.Tensor): input mask (B, 1, seq_len)
+            x_input: Input tensor (B, T, idim) or ((B, T, idim), (B, T, att_dim))
+            masks: Input mask (B, 1, T)
 
         Returns:
-            xs (torch.Tensor): output tensor (B, new_seq_len, odim)
-            masks (torch.Tensor): output mask (B, 1, new_seq_len)
+            x_output: Output tensor (B, sub(T), odim)
+                          or ((B, sub(T), odim), (B, sub(T), att_dim))
+            mask: Output mask (B, 1, sub(T))
 
         """
-        xs = xs.transpose(1, 2).contiguous()
+        if isinstance(x_input, tuple):
+            xs, pos_emb = x_input[0], x_input[1]
+        else:
+            xs, pos_emb = x_input, None
+
+        xs = xs.transpose(1, 2)
         xs = self.tdnn(xs)
 
+        if self.relu:
+            xs = self.rel(xs)
```

Comment:
Maybe use `self.relu_fuc` or just use `F.relu`  like previous code are more easy to understand? I am not sure about it.